app/js/enhancedDate.js

```diff
@@ -1,5 +1,144 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+/*
+ * Carolyn Velez
+ * JSCRIPT 300
+ * 04/14/15
+ */
 
+var enhancedDate = (function iife() {
+
+	// private
+	var myDate;
+
+	// public API
+	return {
+
+		setDate: function setDate(specificDate) {
+			if (specificDate === undefined) {
+				myDate = new Date;
+			} else {
+				myDate = specificDate;
```

Comment:
This works, but keep in mind that objects are passed by reference, not value, when passed in as arguments to a function call. What this means is you've assigned myDate with a reference to an external object. If that external object were modified outside your module, it would modify myDate in your module as well. Better to just create a new Date instance using what they passed in: myDate = new Date(specificDate);